### DISCUSSION TITLE
Control the order of sidebar folders

### DISCUSSION THREAD

**Discussion by pmackay** [lookbook-hq/lookbook#734] [Feature requests/ideas]
Is there any way to configure the order of sidebar folders? So to order by small to large, my preference would be:

- Elements
- Components

but the current alphabetical ordering gives:

- Components
- Elements

**Comment by allmarkedup** [comment]
Hi @pmackay, no I'm afraid at the moment there isn't :-(

It would definitely be a useful feature though and really the only reason it is not in there is because I haven't been able to think of a nice way to implement it.

It's possible to control the order of 'pages' directories by using a prefix on the folder name. But that won't work for previews because it would mess with the Rails autoloading/namespacing mechanism.

Have you got any thoughts on how you'd like to see something like this implemented? I guess it could be done with a configuration array of folder names but it seems pretty ugly to have to keep that manually in sync with the filesystem.

**Comment by allmarkedup** [comment]
Actually, now that you have got me thinking about it again I'm wondering if using a tag annotation (like `@position`) at the module level could work somehow... I might try to explore that a bit when I get a chance :-)

**Comment by pmackay** [comment]
A tag annotation sounds great. I've used `sort_by` with an array of different method calls inside the block to sort against. So could sort by position first, then by name/label?
